src/Compiler/Checking/ConstraintSolver.fs

```diff
@@ -1222,6 +1234,11 @@ and SolveTypeEqualsType (csenv: ConstraintSolverEnv) ndeep m2 (trace: OptionalTr
     let sty1 = stripTyEqnsA csenv.g canShortcut ty1
     let sty2 = stripTyEqnsA csenv.g canShortcut ty2
 
+    let csenv = 
+        match ty1 with 
```

Comment:
This should match sty1 I think

Reply:
sty1 is stripped, and stripping also removes the first "layers of typars" . The IsSupportsNullFlex flag was only on the topmost level, but not deeper.

Other option would be to change the stripping function to carry flags, but that felt to invasive (used in a lot more places.)

Reply:
I'm a little surprised you need to push the flag into csenv.

Reply:
I was trying without it, but was losing it due to abovementioned stripping.

(e.g. a typar whose solution is another typar. Flag sitting on the top, but the constraint was lower in the chain)